posts/specfile-introduction/index.md

```diff
@@ -0,0 +1,141 @@
+---
+title: Introduction to specfile library
+date: 2024-01-12
+authors: nforro
+tags:
+  - specfile
+---
+
+# Introduction to specfile library
+
+Have you ever wanted to make changes in a RPM spec file programatically? _specfile_ library
+has been created for that very purpose. It is a pure Python library that allows you to conveniently
+edit different parts of a spec file while doing its best to keep the resulting changeset minimal
+(no unnecessary whitespace changes etc.).
+
+## Installation
+
+The library is packaged for Fedora, EPEL 9 and EPEL 8 and you can simply instal it with dnf:
+
+```bash
+dnf install python3-specfile
+```
+
+On other systems, you can use pip (just note that it requires RPM Python bindings to be installed):
+
+```bash
+pip install specfile
+```
+
+## Usage
+
+Let's have a look at a few simple examples how to use the library.
+
+### Bumping release
+
+To bump release and add a new changelog entry, we could use the following code:
+
+```python
+from specfile import Specfile
+
+with Specfile("example.spec") as spec:
+    spec.release = str(int(spec.expanded_release) + 1)
+    spec.add_changelog_entry("- Bumped release for test purposes")
+```
+
+Let's take a look at what happens here:
+
+We instantiate `Specfile` class with a path to our spec file and use it as a context manager
+to automatically save all changes upon exiting the context.
+
+We then use `expanded_release` property to get the current value of `Release` tag after macro expansion.
+We assume it is numeric, so we simply convert it to integer, add 1, convert the result back to string
+and assign the new value to `release` property.
+
+:::tip
+
+Note that `release`/`expanded_release` properties exclude dist tag (usually `%{?dist}`) - for convenience,
+it is ignored when reading and preserved unmodified when writing. If that's not what you want, you can use
+`raw_release`/`expanded_raw_release` properties instead.
+
+:::
+
+Finally we add a new changelog entry. We don't specify any other arguments but content,
+so the author is determined automatically using the same procedure as `rpmdev-packager` uses
+and date is set to current day.
+
+### Switching to %autochangelog
+
+To make a switch from traditional changelog to `%autochangelog`, we could do the following:
+
+```python
+import pathlib
+from specfile import Specfile
+
+spec = Specfile("example.spec", autosave=True)
+
+with spec.sections() as sections:
+    entries = sections.changelog[:]
```

Comment:
I was wondering what's going on, since you're not using the `entries` in the scope, and then noticed the line afterwards… I'm really starting to hate this language :D